tasty-query/shared/src/main/scala/tastyquery/reader/tasties/TreeUnpickler.scala

```diff
@@ -1317,19 +1317,39 @@ private[tasties] class TreeUnpickler private (
       reader.readByte()
       SingletonTypeTree(readTerm)(spn)
     case REFINEDtpt =>
-      val spn = span
-      val cls = localCtx.getSymbol[ClassSymbol](reader.currentAddr)
-      reader.readByte()
-      val end = reader.readEnd()
-      val parent = readTypeTree
-      val statements = readStats(end)(using localCtx.withOwner(cls))
-      val refinements = statements.map {
-        case memberDef: RefinementMemberDef =>
-          memberDef
-        case otherDef =>
-          throw TastyFormatException(s"Unexpected member $otherDef in refinement type")
-      }
-      RefinedTypeTree(parent, refinements, cls)(spn)
+      /* It is possible to find SHAREDterm's referencing REFINEDtpt's.
```

Comment:
I thought the point of SHAREDterm and SHAREDtype is that it should reuse whatever was read the first time, and never re-parse

Reply:
You might think so, but even dotc disagrees. It truly caches Types, but it re parses Trees and TypeTrees. And in this case, it will in fact re parse trees defining the same symbols, creating something wrong in the compiler. 🤷‍♂️